### PR TITLE
SEDB-43 setup GraphQL schema in server

### PR DESCRIPTION
**Pull request by Amym123**
- added some queries and mock data to play around with.

Todo: 
- add models for mongoose
- hook up mongo with graphql
- add resolvers

image

### CODE REVIEW

File: server/src/typeDefs.js [hunk 1]
```diff
@@ -1,7 +1,106 @@
...
+    biography: String
+    avatar: String
+    gender: String
+    preferences: Preferences
+    measurements: Measurements
+    inquiryIds: [ID]
+  }
+
+  type Styler {
+    userId: ID
```
Comment (na-wu) [inline server/src/typeDefs.js:28]: Styler `userId` would also be non-nullable right? as with stylee

**Comment by na-wu** [comment]
Looks good!

Just 1 concern regarding the fields that are supposed to be required. Aside from primary keys such as IDs, would other fields need to be required by default as well? (Basically in my head I just imagined most of the fields are required except for images and a few extra metadata

File: server/src/typeDefs.js (hunk 1 above)
Comment (Amym123) [reply server/src/typeDefs.js]: right! ty!

File: server/src/typeDefs.js [hunk 2]
```diff
@@ -1,7 +1,106 @@
...
+    userId: ID!
+    accountCreationDate: Date
+    username: String
+    email: String
+    firstName: String
+    lastName: String
+    birthdate: String
+    location: Location
+    biography: String
+    avatar: String
```
Comment (jameskanCa) [inline server/src/typeDefs.js:37]: Avatar can be changed to type image instead of type string. (name is fine)

File: server/src/typeDefs.js [hunk 3]
```diff
@@ -1,7 +1,106 @@
...
```
Comment (jameskanCa) [inline server/src/typeDefs.js:20]: Same comment regarding the type. (image not string)

File: server/src/exampleQuery.txt [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (tsangela) [inline server/src/exampleQuery.txt:1]: this can be in server README
easier to access and doesn't involve pushing up the example to the actual codebase

**Comment by tsangela** [comment]
Feel free to resolve the comments when you've pushed up the changes. You can probably merge after moving the example to README.

**Comment by jameskanCa** [comment]
LGTM

File: server/src/exampleQuery.txt (hunk 4 above)
Comment (na-wu) [reply server/src/exampleQuery.txt:1]: +1

**Pull request merged**